src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -63,7 +63,8 @@ public static ConsoleColor BackgroundColor
 
         public static void ResetColor()
         {
-            if (!ConsoleOutIsTerminal)
+            // We only want to reset colors if we're targeting a TTY device
+            if (Console.IsOutputRedirected)
```

Comment:
> I thought we intentionally do this on Unix?

We do, for the color changing stuff.  We didn't have IsXxRedirected before, so there was no intenionality about that :wink:.

> Why do we make the Color behave this way? On windows even Color is based on standard input, output and error.

Because on Unix we change the color by actually writing to the output stream.  If it's not going to something that knows how to interpret that output and remove the output from what's displayed, you end up with lots of funky characters in the output stream.  You could easily see this before with Jenkins, where these ended up on every line of the output log where the color was changed by xunit.